akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/ProtobufProtocolConversions.scala

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (C) 2009-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.grpc.internal
+
+import akka.annotation.InternalApi
+import akka.persistence.query.NoOffset
+import akka.persistence.query.Offset
+import akka.persistence.query.TimestampOffset
+import akka.persistence.query.typed.EventEnvelope
+import akka.persistence.typed.PersistenceId
+import akka.projection.grpc.consumer.ConsumerFilter
+import akka.projection.grpc.internal.FilterStage.Filter
+import akka.projection.grpc.internal.proto.EntityIdOffset
+import akka.projection.grpc.internal.proto.Event
+import akka.projection.grpc.internal.proto.ExcludeEntityIds
+import akka.projection.grpc.internal.proto.ExcludeRegexEntityIds
+import akka.projection.grpc.internal.proto.ExcludeTags
+import akka.projection.grpc.internal.proto.FilterCriteria
+import akka.projection.grpc.internal.proto.FilteredEvent
+import akka.projection.grpc.internal.proto.IncludeEntityIds
+import akka.projection.grpc.internal.proto.IncludeRegexEntityIds
+import akka.projection.grpc.internal.proto.IncludeTags
+import akka.projection.grpc.internal.proto.IncludeTopics
+import akka.projection.grpc.internal.proto.PersistenceIdSeqNr
+import akka.projection.grpc.internal.proto.RemoveExcludeEntityIds
+import akka.projection.grpc.internal.proto.RemoveExcludeRegexEntityIds
+import akka.projection.grpc.internal.proto.RemoveExcludeTags
+import akka.projection.grpc.internal.proto.RemoveIncludeEntityIds
+import akka.projection.grpc.internal.proto.RemoveIncludeRegexEntityIds
+import akka.projection.grpc.internal.proto.RemoveIncludeTags
+import akka.projection.grpc.internal.proto.RemoveIncludeTopics
+import akka.projection.grpc.producer.scaladsl.EventProducer.Transformation
+import com.google.protobuf.timestamp.Timestamp
+
+import java.time.Instant
+import scala.collection.immutable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Success
+
+/**
+ * INTERNAL API
+ *
+ * Common conversions between protobuf protocols and Akka APIs
+ */
+@InternalApi
+private[akka] object ProtobufProtocolConversions {
```

Comment:
this was pretty much just moved code, or anything that needs a closer review?

Reply:
Just extracted/moved so it can be shared with the existing grpc projections.